### DISCUSSION TITLE
how can I implement dynamic titles using lit-ssr?

### DISCUSSION THREAD

**Discussion by wencan** [lit/lit#4861] [General]
### Should this be an RFC?

- [x] This is not a substantial change

### Which package is this a feature request for?

SSR (@lit-labs/ssr)

### Description

My webpage consists of several Lit components. One of these components holds the core content of the webpage. This core component fetches the main content through an API and extracts the title from it.
Currently, my solution is: after the component calls the API, it dispatches a CustomEvent carrying the title. An external script listens for this event and then updates the document.title.
Now, how can I implement dynamic titles using lit-ssr?

### Alternatives and Workarounds

Nothing

**Comment by sorvell** [comment]
Sorry, this isn't currently supported. 

There are roughly 2 changes needed: (1) support for events ([PR in review](https://github.com/lit/lit/pull/4755)), (2) a setting to turn off streaming: needed since the `<title>` would have to wait for the event from the element before emitting. The streaming setting has been discussed but is not currently planned.

Alternatively, you'd need to do something custom like have your server code separately call your title API and emit the title element.

**Comment by wencan** [comment]
@sorvell 
I am developing an SPA website. 
I have been wondering whether using events to update the title is a good solution. 
Do you have any better suggestions?

**Comment by justinfagnani** [comment]
You can use server-only templates to render the head and `<title>` tag on the server, then use client-only code to update the title dynamically. It sounds like your core component could just update the title directly, rather than dispatch an event, but it should be workable either way.
